### PR TITLE
mailer smtp integration, priority headers, cleanup / refactoring, fixed broken logging

### PR DESCRIPTION
**Pull request by JohnTheodore**
- logger was imported on each individual file, and sometimes not imported (so no logging happened in certain instances). I refactored the logging to the logger levels are set once for the mailer in periodic.py, and that sets it everywhere else, and the one logger object is passed around where needed.

- you can get flags, exclamation marks or stars on your emails depending on the client if you set the x-priority header, this is now available. (it required switching to ses send_raw_email).

- smtp support is now emailed (send_raw_email from ses) but use the same MimeText email format as python smtplib which is convenient. We have private DLs in exchange, that require auth to send emails to. SES doesn't support this auth. Our lambda mailer will have access to our private VPC to talk to our smtp relay also: https://docs.aws.amazon.com/lambda/latest/dg/vpc.html#vpc-configuring .. so this is great. Now we can send tagging and other support emails to private DLs, which we couldn't before. also external emails (from SES) get tagged as [EXT] by our exchange, so not those don't get tagged like that anymore, since they come from inside (so to speak). smtp defaults to using ssl.

- there were parts of the code that weren't referenced by anything (dead code). for instance there were two ses send mail methods in the code base (now there is one). I did various cleanup and removing unreferenced variables, etc.

- the mailer code was confusing, 'Worker', 'Processor', process_data_message, send_resource_set_message, etc. I renamed from variables, functions and classes after I learned what they do. process_data_message -> process_sqs_messsage, Worker -> MailerSqsQueueProcessor. etc. One of the functions was over 50 lines, so I broke it up into smaller logical pieces so it's easier to think about.

Let me know if you want me to change anything!

### CODE REVIEW

**Comment by kapilt** [comment]
thanks, still digging through this. 

fwiw, logger importing separately is fine, thats how python logging works, it creates singletons behind the scenes, whether a message shows up is more about which log channel name is used and where in the hierarchy it is, and the relevant config on the hierarchy (the stdlib logging is kinda of log4j inspired for better or worse), but explicitly passing around log channels as params is a red-herring as its not the interface for that library.

**Comment by JohnTheodore** [comment]
yes, logging is a singleton, but if you set the level in multiple files it's likely to cause confusing behavior if people set it to different levels. also you can more easily transition to setting the log level in a config file on start with this method. (if someone wanted that).

I will note I was confused by worker.py and why it didn't output the logging information. I set a debugger in there, and manually did log.info('test'), and nothing printed out. The log object wasn't configured, and I believe if it's not configured (basicConfig) logs go nowhere? after my change worker.py prints to stdout.

**Comment by kapilt** [comment]
Overall, this looks really good, thanks!

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 *.pyc
 *.egg-info/
+*.swp
```
Comment (JohnTheodore) [inline .gitignore]: The vim editor makes swp files when you're editing a file and haven't saved yet, it's a buffer for the editor. sometimes it peppers these files around, so I want to make sure I don't accidentally check them in as commits.

**Comment by JohnTheodore** [comment]
https://github.com/capitalone/cloud-custodian/issues/706

**Pull request merged**